content/cookbook/05-node/54-mcp-tools.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: Model Context Protocol (MCP) Tools
+description: Learn how to use MCP tools with the AI SDK and Node
+tags: ['node', 'tool use', 'agent', 'mcp']
+---
+
+# MCP Tools
+
+The AI SDK supports Model Context Protocol (MCP) tools by offering a lightweight client that exposes a `tools` method for retrieving tools from a MCP server. After use, the client should always be closed to release resources.
+
+```ts
+import { experimental_createMCPClient, generateText } from 'ai';
+import { openai } from '@ai-sdk/openai';
+
+let clientOne;
+let clientTwo;
+
+try {
+  // Initialize an MCP client to connect to a `stdio` MCP server:
+  clientOne = await experimental_createMCPClient({
+    transport: {
+      type: 'stdio',
+      command: 'node server.js',
+    },
+  });
+
+  // Alternatively, you can connect to a Server-Sent Events (SSE) MCP server:
+  clientTwo = await experimental_createMCPClient({
+    transport: {
+      type: 'sse',
+      url: 'http://localhost:3000/sse',
+    },
+  });
+
+  const toolSetOne = await clientOne.tools();
+  const toolSetTwo = await clientTwo.tools();
+
+  const tools = {
+    ...toolSetOne,
+    ...toolSetTwo,
+  };
```

Comment:
not sure we need to do something here, but a potential error scenario is that `toolSetTwo` silently overrides tools from `toolSetOne` when there  naming conflicts